### ISSUE TITLE
After switching controllers, the node names and locations are attributed to the wrong devices

### ISSUE THREAD

**Issue by AlCalzone** [zwave-js/zwave-js-ui#3817]
![grafik](https://github.com/user-attachments/assets/ed62e3f9-cb7e-4862-bb6c-3d30b9f539ac)

I previously had another stick connected with 3 nodes, using different IDs 2, 3 and 4, which had these names and locations set. After switching sticks, the names and locations persist in the UI, but they belong to the wrong devices.

**Comment by robertsLando** [comment]
yeah this is expected, on init I load nodes.json and apply it to the existing devices, this is useful in case you restore your backup on a different controller to keep nodes names in sync. The alternative would be to store nodes along with the homeID. I accept suggestions here as this is not a very common use case and we should consider pro-cons of changing the actual behaviour

**Comment by AlCalzone** [comment]
I think the node names should be stored separately for each home ID

**Comment by robertsLando** [comment]
> Or you could use the properties from the node instance to have the driver handle it?

I actually pass down to the driver the name/location if I see them are different to the ones stored on nodes.json

That file exist for very old reasons as with OZW node name and location was not working everytime so I decided to keep them stored there. We could also decide to simply drop name/loc from that file as it is alsready stored on your json backup. 

I also store home assistant custom devices on that file and that will remain

**Comment by AlCalzone** [comment]
In any case, nodes belonging to a different home ID are very likely not the same devices. Especially if the Node ID also changed.

**Comment by AlCalzone** [comment]
@robertsLando I still think we should tackle this. It is extremely confusing that devices get some unrelated names just because they have the same node id another device previously had

**Comment by robertsLando** [comment]
will take a look at this next week, the fix will add the homeid to the nodes file
